js/index.js

```diff
@@ -30,25 +33,72 @@ app.controller("IndexController", ["$scope", "$http", function($scope, $http) {
     var urlString = "http://nominatim.openstreetmap.org/reverse?format=json&lat=" +
     latlng.lat + "&lon=" +
     latlng.lng + "&zoom=18&addressdetails=1";
-    $http.get(urlString).then(addMarker);
+    $http.get(urlString).then(addMarker,errorMessage);
   }
 
   function addMarker(response) {
+
     var marker = {
       lat: parseFloat(response.data.lat),
       lng: parseFloat(response.data.lon),
-      message: "New entry",
+      title: "New entry",
       dueDate: new Date(),
       address: response.data.display_name
     };
 
-    $scope.markers.push(marker);
+
+
     $scope.currentMarker = marker;
     $("#modal").modal('show');
+    $("#modal").on('click','#btnClose',function(){
+      //console.log(marker[]);
+      $http.post(serverUrl,marker).then(postSuccess,errorMessage);
+      marker = {};
+    })
   }
 
+
+  $scope.remove = function(index) {
+    if (index != -1) {
+      var id = $scope.markers[index]._id;
+      console.log(id);
+      $http.delete(serverUrl+id).then(function(){
+        $http.get(serverUrl).then(loadTodos,errorMessage);
+      },errorMessage);
+    }
+  }
+  function loadTodos(response){
+    //console.log(response);
+    if(response.data.length>0){
+      console.log(response.data);
+      $scope.markers = response.data;
+    }
+    else{
+      $scope.currentMarker = [];
+    }
+  }
+  function postSuccess(response){
+    //console.log(response);
+    $http.get(serverUrl).then(loadTodos,errorMessage);
+  }
+  function errorMessage(error){
+      console.log(error);
```

Comment:
I think everywhere else we use 2 spaces for indentation. Should this be 2 as well?